sopel/plugins/rules.py

```diff
@@ -482,7 +482,7 @@ def last_time(self) -> Optional[datetime.datetime]:
         """Last recorded start/end time for the associated rule"""
         # detect if we just started something or if it ended
         last_time = self.started_at
-        if self.ended_at and self.started_at < self.ended_at:
+        if self.ended_at and (self.started_at and self.started_at < self.ended_at):
```

Comment:
Blegh, I totally understand this fix in context of the PR, but I don't like the look of it and feel like it would trip me up if I didn't have that context in my head while reading it.

After staring at it for a few minutes, the best alternative spelling I can think of is also kind of dissatisfying because it involves repetition:

```python
if not self.ended_at or not self.started_at:
    last_time = self.started_at
# else both are not-None, mypy narrows that properly
elif self.started_at < self.ended_at:
    last_time = self.ended_at
else:
    last_time = self.started_at    
```

Reply:
It's the fix I'm least happy about in this patch, to be sure. Thinking I'll leave it for a bit and see if @Exirel weighs in with a stronger preference, or a third alternative. (Or maybe I'll have another idea? Who knows!)

Reply:
Any version will come up with will need to ensure that `self.started_at` is not `None` anyway.

Here is another version:

```
if (self.started_at and self.ended_at) and (self.started_at < self.ended_at):
        return self.ended_at

return self.started_at
```

And another one:

```
times = sorted(t for t in [self.started_at, self.ended_at] if t)
return times[-1] if times else None
```

Reply:
(but no, no strong preference here, your version is perfectly fine by me and does exactly what it should, maybe we could get rid of the `last_time` intermediate variable tho)

Reply:
> Here is another version:
> 
> ```python
> if (self.started_at and self.ended_at) and (self.started_at < self.ended_at):
>         return self.ended_at
> 
> return self.started_at
> ```

I think this version is the best balance of brevity + clarity, and as a bonus it eliminates the local variable. 🚀

Reply:
Agreed, I like that version